addons/binding/org.openhab.binding.vera2/src/main/java/org/openhab/binding/vera2/handler/VeraDeviceHandler.java

```diff
@@ -0,0 +1,584 @@
+                    acceptedItemType = "Number";
+                    break;
+                case 17: // Temperature Sensor
+                    id = SENSOR_TEMPERATURE_CHANNEL;
+                    acceptedItemType = "Number";
+                    break;
+                case 18: // Light Sensor
+                    id = SENSOR_LUMINOSITY_CHANNEL;
+                    acceptedItemType = "Number";
+                    break;
+                case 19: // TODO Z-Wave Interface
+                    logger.warn("TODO: Z-Wave Interface: {}, {}, {}, {}", device.id, device.name, device.category,
+                            device.categoryName);
+                    break;
+                case 20: // TODO Insteon Interface
+                    logger.warn("TODO: Insteon Interface: {}, {}, {}, {}", device.id, device.name, device.category,
+                            device.categoryName);
+                    break;
+                case 21: // Power Meter
+                    id = SENSOR_ENERGY_CHANNEL;
+                    acceptedItemType = "Number";
+                    break;
+                case 22: // TODO Alarm Panel
+                    logger.warn("TODO: Alarm Panel: {}, {}, {}, {}", device.id, device.name, device.category,
+                            device.categoryName);
+                    break;
+                case 23: // TODO Alarm Partition
+                    logger.warn("TODO: Alarm Partition: {}, {}, {}, {}", device.id, device.name, device.category,
+                            device.categoryName);
+                    break;
+                case 24: // TODO Siren
+                    logger.warn("TODO: Siren: {}, {}, {}, {}", device.id, device.name, device.category,
+                            device.categoryName);
+                    break;
+                case 25: // TODO Weather
+                    logger.warn("TODO: Weather: {}, {}, {}, {}", device.id, device.name, device.category,
+                            device.categoryName);
+                    break;
+                case 26: // TODO Philips Controller
+                    logger.warn("TODO: Philips Controller: {}, {}, {}, {}", device.id, device.name, device.category,
+                            device.categoryName);
+                    break;
+                case 27: // TODO Appliance
+                    logger.warn("TODO: Appliance: {}, {}, {}, {}", device.id, device.name, device.category,
+                            device.categoryName);
+                    break;
+                case 28: // UV Sensor
+                    id = SENSOR_ULTRAVIOLET_CHANNEL;
+                    acceptedItemType = "Number";
+                    break;
+                default:
+                    logger.warn("Unknown device type: {}, {}, {}, {}", device.id, device.name, device.category,
+                            device.categoryName);
+            }
+
+            // If at least one rule could mapped to a channel
+            if (!id.equals("")) {

```

Comment:
Position literals first in String comparisons - that way if the String is null you won't get a NullPointerException, it'll just return false. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)